test/appium/tests/atomic/dapps_and_browsing/test_deep_links.py

```diff
@@ -19,12 +20,80 @@ def test_open_public_chat_using_deep_link(self):
         chat_name = sign_in_view.get_public_chat_name()
         sign_in_view.send_as_keyevent('https://get.status.im/chat/public/%s' % chat_name)
         sign_in_view.confirm()
-        open_button = sign_in_view.element_by_xpath('//*[@text="Open in Status"] | //*[@content-desc="Open in Status"]')
-        open_button.wait_for_visibility_of_element()
-        open_button.click()
+        sign_in_view.open_in_status_from_external_browser_button.click()
         sign_in_view.sign_in()
         chat_view = sign_in_view.get_chat_view()
         try:
             assert chat_view.user_name_text.text == '#' + chat_name
         except (AssertionError, NoSuchElementException):
             pytest.fail("Public chat '%s' is not opened" % chat_name)
+
+    @marks.testrail_id(5441)
+    @marks.medium
+    def test_open_user_profile_using_deep_link(self):
+        sign_in_view = SignInView(self.driver)
+        sign_in_view.create_user()
+        self.driver.close_app()
+        start_web_browser(self.driver)
+        sign_in_view.send_as_keyevent('https://get.status.im/user/%s' % basic_user['public_key'])
+        sign_in_view.confirm()
+        sign_in_view.open_in_status_from_external_browser_button.click()
+        sign_in_view.sign_in()
+        chat_view = sign_in_view.get_chat_view()
+        for text in basic_user['username'], 'Add to contacts', 'Send transaction':
+            if not chat_view.element_by_text(text).scroll_to_element():
+                pytest.fail("User profile screen is not opened")
+
+
+    @marks.testrail_id(5442)
+    @marks.medium
+    def test_open_dapp_using_deep_link(self):
+        sign_in_view = SignInView(self.driver)
+        sign_in_view.create_user()
+        self.driver.close_app()
+        start_web_browser(self.driver)
+        dapp_name = 'simpledapp.eth'
+        sign_in_view.send_as_keyevent('https://get.status.im/browse/%s' % dapp_name)
```

Comment:
A separate method should be added, all 4 lines are repeated in each test the only diff is all tests

Reply:
Fixed.